server/main.go

```diff
@@ -0,0 +1,221 @@
+// see https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/examples/server/main.go#L30
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"runtime/debug"
+	"syscall"
+
+	grpcprom "github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/recovery"
+	pb "github.com/jonas27/ramp-up-k8s-operator/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"go.opentelemetry.io/otel"
+	stdout "go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/exp/slog"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	exitFail             = 1
+	serverTimeoutSeconds = 10
+)
+
+// interceptorLogger adapts slog logger to interceptor logger.
+// This code is simple enough to be copied and not imported.
+// see https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/interceptors/logging/examples/slog/example_test.go
+func interceptorLogger(l *slog.Logger) logging.Logger { //nolint:ireturn
+	return logging.LoggerFunc(func(ctx context.Context, lvl logging.Level, msg string, fields ...any) {
+		l.Log(ctx, slog.Level(lvl), msg, fields...)
+	})
+}
+
+func main() {
+	log := slog.New(slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{AddSource: true}))
+
+	if err := run(os.Args, log); err != nil {
+		log.Error(err.Error())
+		os.Exit(exitFail)
+	}
+}
+
+type server struct {
+	pb.UnimplementedCharacterCounterServer
+}
+
+func (s *server) CountCharacters(_ context.Context, req *pb.CountCharactersRequest) (*pb.CountCharactersResponse, error) { //nolint:lll
```

Comment:
There seem to be some communication problems. Where did you get this proto definition from? The rampup challenges document doesn't state this anywhere

Reply:
I wanted to do something interesting ;)

Reply:
returns binary string

Reply:
I can also change it to really use strings, but that takes a little more changes